.travis.yml

```diff
@@ -1,11 +1,27 @@
-language: scala
 jdk: openjdk8
-
+language: scala
 scala:
   - 2.12.2
+  
+# https://www.scala-sbt.org/1.0/docs/Travis-CI-with-sbt.html
+cache:
+  directories:
+    - $HOME/.ivy2/cache
+    - $HOME/.sbt
+    - /opt/iv2
 
-before_script:
-  - sudo ./get_and_build_deps.sh
+before_cache: 
+  # Cleanup the cached directories to avoid unnecessary cache updates
+  - rm -fv $HOME/.ivy2/.sbt.ivy.lock
+  - find $HOME/.ivy2/cache -name "ivydata-*.properties" -print -delete
+  - find $HOME/.sbt        -name "*.lock"               -print -delete
+  
+  - sudo rm -fv /opt/ivy2/.sbt.ivy.lock
+  - sudo find /opt/ivy2/cache -name "ivydata-*.properties" -print -delete
 
+install:
+  - sudo ./get_and_build_deps.sh >/dev/null
```

Comment:
This is why it takes so long... right now, we're building all of the dependencies from scratch. 

I think 15 minutes is reasonable for public CI testing. It should be much faster locally since you would already have everything installed.